### PR TITLE
Adds a dual-ended telescopic baton!

### PR DESCRIPTION
**Pull request by Kmc2000**
Refer to issue https://forums.yogstation.net/index.php?threads/double-bladed-telescopic-baton.16005/#post-142080

### Intent of your Pull Request
Video:
https://drive.google.com/file/d/1KN0x7usBtIqqK_YXNs4oiZ8yrS7n_dRH/view?usp=sharing

So this adds a dual ended telebaton, it's very strong but situational as well as being hard to spam. It will disarm people on any given hit but in order to actually stun them you must perform a 4 step combo move.

Steps below:

Harm intent and hit:
Gives stamina damage

Harm intent and hit:
Head damage to enemy as well as slight dizzying effect (comparable to peaceborg spray)

Disarm intent and hit:
Disarms the enemy with a small stun

Grab intent and hit:
Gives a medium stun to the enemy and attempts to launch you towards them, if you hit them they are stunned for 9 seconds (Just over a stunbaton), pretty much guaranteeing a kill, but if you fail you are stunned and the item is dropped, this will probably happen if someone walks in your way, so be careful on that last hit!.

Antags,
If you kill two heads you can become a force to be reckoned with.

Feedback welcome, especially on the grammar standpoint.

#### Changelog

:cl: Kmc2000
rscadd: Added a double ended telescopic baton! craft one today and get combo-ing.
/:cl:

### CODE REVIEW

**Comment by Alexandria404** [comment]
Nukeops, if you can get two heads to contribute to this monstrosity then you can take nukeops out of the game (with a lot of luck and some nice timing / dodging)

You mean you get shot and killed?

**Comment by Kmc2000** [comment]
The stun is very powerful on it, so if you're quick with your reactions you can score the kill with it.

Also
This is yogstation
>expecting nukeops to have aim

**Comment by Ratvar** [comment]
Changelings and traitors/ops (+ chemist on drugs) will adrenal out of stun if they fail, and grab takes no time with altclicking
Double baton just will become 20second stunbaton
Add carp scroll gang into team
Add future slip scroll for traitor janitor into team?


Make grab the agressive/tripple one, maybe?

**Comment by Nickvr628** [comment]
Can't RnD print telescopic batons? Infinite Darth Mauls?

**Comment by Kmc2000** [comment]
Grab INTENT.
Not a normal grab
So yeah you still need to cycle through your intents.

**Comment by Kmc2000** [comment]
>Nick

I didn't know that was a thing
:hyperlul:

**Comment by Alexandria404** [comment]
cycle through intents? I have them on 1-4

**Comment by Kmc2000** [comment]
Dirty powergamer.

So instead of clicking intents you still have to press buttons.

File: code/game/objects/items/weapons/melee/misc.dm [hunk 1]
```diff
@@ -131,6 +131,136 @@
...
+			switch(streak)
+				if(1)
+					if(user.a_intent == "harm")
+						H.visible_message("<span class='warning'>[user] punts their [src] into [H]'s stomach, winding them!</span>")
+						H.adjustStaminaLoss(rand(5,10))
+						H.Stun(1)
+						return
+					else
+						current_target = null
+						streak = 0
```
Comment (ThatLing) [inline code/game/objects/items/weapons/melee/misc.dm]: Why do this a bunch of times instead of creating a proc that resets the target?

**Comment by ghost** [comment]
why

**Comment by Alexandria404** [comment]
Nevermind this could use some balancing i guess.

**Comment by Kmc2000** [comment]
What balancing do you suggest?

**Comment by ghost** [comment]
can't wait to perma knockout a ling for easy cremation no fort save because it can't use abilities when unconcious 

great item

10/10 would fucking hate it if i could actually go on the server

i still hate it in principle

**Comment by Amelia0010** [comment]
So we just playtested this. Jesus fuck it's overpowered, what the fuck KMC

**Comment by Amelia0010** [comment]
I should just fucking close this now

**Comment by Moja10** [comment]
God if you actually made something stupidly stronger than a desword for such a beautiful idea then i'm ashamed

**Comment by Zostroll** [comment]
I have a telescopic baton you extent it by flicking it away from you. There would be no way to open both sides of one at the same time. Video to demonstrate https://youtu.be/ga2oRUl_n50?t=3m20

**Pull request closed without merging**